lib/ood_core/batch_connect/templates/vnccontainer.rb

```diff
@@ -0,0 +1,248 @@
+require "ood_core/refinements/hash_extensions"
+
+module OodCore
+  module BatchConnect
+    class Factory
+      using Refinements::HashExtensions
+
+      # Build the VNC template from a configuration
+      # @param config [#to_h] the configuration for the batch connect template
+      def self.build_vnccontainer(config)
+        context = config.to_h.symbolize_keys.reject { |k, _| k == :template }
+        Templates::VNCContainer.new(context)
```

Comment:
I get the feeling we should raise an error here. This isn't likely to work out of the box, so I'm inclined to fail fast.

I'm not 100% sure about the language here, but this seems as good as any. Feel free to improve if if you have something better.
```suggestion
        context = config.to_h.symbolize_keys.reject { |k, _| k == :template }
        
        unless context.key?(:container_path)
          raise JobAdapterError, "You are missing the configuration 'container_path' for a vnc_container template."
        end

        Templates::VNCContainer.new(context)
```